### PR TITLE
[SPARK-15715][SQL] Fix alter partition with storage information in Hive

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

This command didn't work for Hive tables. Now it does:

```
ALTER TABLE boxes PARTITION (width=3)
    SET SERDE 'com.sparkbricks.serde.ColumnarSerDe'
    WITH SERDEPROPERTIES ('compress'='true')
```
## How was this patch tested?

`HiveExternalCatalogSuite`

### CODE REVIEW

**Comment by cloud-fan** [comment]
LGTM

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 1]
```diff
@@ -779,18 +780,29 @@ private[hive] class HiveClientImpl(
     hiveTable
   }
 
-  private def toHiveViewTable(view: CatalogTable): HiveTable = {
```
Comment (andrewor14) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: not used

**Comment by cloud-fan** [comment]
LGTM, pending jenkins

**Comment by andrewor14** [comment]
Merging into master 2.0 thanks.

**Pull request closed without merging**